src/test/java/ch/heigvd/gen/monopoly/squares/TaxIncomeSquareTest.java

```diff
@@ -0,0 +1,47 @@
+package ch.heigvd.gen.monopoly.squares;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class TaxIncomeSquareTest extends SquareTest {
+
+  @Override
+  protected Square provideSquare() {
+    return new TaxIncomeSquare("Tax income");
+  }
+
+  @Test
+  public void testPlayerWithNoMoneyRemainsWithNoMoney() {
+    player.reduceCash(player.getNetWorth());
+    assertEquals(0, player.getNetWorth());
+
+    square.landedOn(player);
+    player.setLocation(square);
+
+    assertEquals(0, player.getNetWorth());
+  }
+
+  @Test
+  public void testPlayerWithLessThanMinimumLosesTenPercent() {
+    player.reduceCash(500);
+    assertEquals(1000, player.getNetWorth());
+
+    square.landedOn(player);
+    player.setLocation(square);
+
+    assertEquals(900, player.getNetWorth());
+  }
+
+  @Test
+  public void testPlayerWithLoadsOfMoneyLosesTwoHundredDollars() {
+    // Who wants to be a millionaire ?
```

Comment:
Prem Kumar:
Final question for twenty million rupees, and he's smiling. I guess you know the answer.

Jamal:
Do you believe it, I don't!

Prem Kumar:
You don't? So you take the ten million and walk?

Jamal:
No. I'll play.